.github/workflows/ci.yml

```diff
@@ -44,6 +44,57 @@ jobs:
         uses: ./.github/actions/paths-filter
         id: filter
 
+  dummy:
+    needs: [detect-changes]
+    runs-on: gcp-core-2-default
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - run: sleep 300
+      - name: Observe build status
+        if: always()
+        continue-on-error: true
+        uses: ./.github/actions/observe-build-status
+        with:
+          build_status: ${{ job.status }}
+          secret_vault_address: ${{ secrets.VAULT_ADDR }}
+          secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
+
+  dummy2:
+    if: always()
+    needs: [dummy]
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - env:
+          GH_TOKEN: ${{ github.token }}
+          BUILD_BASE_REF: "${{ github.base_ref && format('refs/heads/{0}', github.base_ref) || github.event.merge_group.base_ref }}"
+        run: |
+          gh api -X GET "repos/camunda/camunda/actions/runs/$GITHUB_RUN_ID/attempts/$GITHUB_RUN_ATTEMPT/jobs?per_page=100" --jq '.jobs[] | select(.conclusion=="failure") .id' | while read job_id; do
+            echo "Processing job: $job_id"
+            if gh api -X GET "/repos/camunda/camunda/check-runs/$job_id/annotations" --jq '.[] | select(.message | contains("lost communication with the server") or contains("runner has received a shutdown signal"))' | grep -q message; then
```

Comment:
:thought_balloon: Oh, cool! I was always getting the check suite and from it the run IDs. I didn't know you could just use the job ID directly for the runs themselves!

(ref: https://camunda.slack.com/archives/C03B0F6FUQG/p1712763908733729)

Reply:
> didn't know you could just use the job ID directly for the runs themselves!

So much this! Got to this thanks to camgpt and also put an extra comment in my notes saying "https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-run-annotations → `check_run_id == job_id!`" 😀